acceptancetests/assess_updateseries.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python
+""" Assess updating the series of existing and future juju units.
+
+Bootstrap juju and check that juju handles the series updates of applications
+appropriately and assists getting the juju agents back up and running if the
+service changes.
+
+  - Bootstraps a controller
+  - Deploy two units of a charm using trusty as a series
+  - Update each to Xenial, LTS version, reboot one but not the other.
+  - Run jujud-updateseries command twice on rebooted unit, using
+    --start-agents the second time, agents should start.
+  - Run jujud-updateseries command twice on non-rebooted unit, using
+    --start-agents the second time, agents won't be running.
+  - Reboot the unit which hadn't be rebooted already, agents should start.
+  - Run juju update-series to update the series of both machine in juju status
+  - Run juju update-series to update the series the charm uses to deploy.
+  - Deploy a 3rd unit of the charm, ensure is using the new series.
+"""
+
+from __future__ import print_function
+
+import argparse
+import logging
+import subprocess
+import sys
+
+from deploy_stack import (
+    BootstrapManager,
+)
+from jujucharm import (
+    local_charm_path,
+)
+from utility import (
+    add_basic_testing_arguments,
+    configure_logging,
+    JujuAssertionError,
+    wait_for_port,
+)
+
+__metaclass__ = type
+
+log = logging.getLogger("assess_updateseries")
+charm_app = 'dummy-source'
+
+expectedStartAgents = "started jujud-unit-{}-{} service\n" + \
+                      "started jujud-machine-{} service\n" + \
+                      "all agents successfully restarted"
+
+
+def assess_jujud_updateseries(client, args):
+    """ Tests juju-updateseries --to-series <series> --from-series <series>
+
+    Using the origin trusty unit, run do-release-upgrade
+    Run juju-updateseries on unit
+    Run juju-updateseries on unit --start-agents, it should fail
+    Reboot the unit, verify agents start.
+
+    Add a new trusty unit,
+    run do-release-upgrade and reboot
+    Run juju-updateseries on unit
+    Run juju-updateseries on unit with --start-agents
+    Verify agents start.
+    """
+    upgrade_release(client, '0', args.from_series, args.to_series, 'after')
+    unit_num = client.get_status().get_service_unit_count(charm_app)
+    client.juju('add-unit', (charm_app))
+    client.wait_for_started()
+    upgrade_release(client, unit_num, args.from_series, args.to_series,
+                    'before')
```

Comment:
Being picky, newlines to confirm to the 80-column width should happen at the earliest bracket (unless doing so makes the code unreadable).
this would become: 

```python
upgrade_release(
    client, unit_num, args.from_series, args.to_series, 'before')
```

Reply:
updated